CompilerSource/compiler/components/write_object_data.cpp

```diff
@@ -17,19 +17,18 @@
 *** with this code. If not, see <http://www.gnu.org/licenses/>
 **/
 
-#include "settings.h"
-#include "parser/parser.h"
 #include "backend/GameData.h"
 #include "compiler/compile_common.h"
 #include "event_reader/event_parser.h"
 #include "general/parse_basics_old.h"
-#include "settings.h"
 #include "languages/lang_CPP.h"
+#include "parser/parser.h"
```

Comment:
This is a very large review, and it's made more difficult by the automatic formatting changes being thrown in. Can you remove these from the PR, or create a separate PR to check them in and then rebase yours onto it?

In general, formatting and refactors are best kept separate from functional changes as it improves reviewability and makes it easier to tell why a change was made later (e.g. when picking through the [blame layer](https://git-scm.com/docs/git-blame)).